src/com/xilinx/rapidwright/interchange/DeviceResourcesWriter.java

```diff
@@ -629,8 +647,24 @@ public static void writeAllSiteTypesToBuilder(Design design, Device device, Devi
             for (int j=0; j < allSitePIPs.length; j++) {
                 DeviceResources.Device.SitePIP.Builder spBuilder = spBuilders.get(j);
                 SitePIP sitePIP = allSitePIPs[j];
-                spBuilder.setInpin(allBELPins.getIndex(sitePIP.getInputPin()));
-                spBuilder.setOutpin(allBELPins.getIndex(sitePIP.getOutputPin()));
+                BELPin inputPin = sitePIP.getInputPin();
+                spBuilder.setInpin(allBELPins.getIndex(inputPin));
+                BELPin outputPin = sitePIP.getOutputPin();
+                spBuilder.setOutpin(allBELPins.getIndex(outputPin));
+
+                // Write out delay of SitePIP
+                if (intrasiteAndLogicDelayModel != null) {
+                    BELPin frBelPin = inputPin.getSourcePin();
+                    String frBelPinString = (frBelPin.isSitePort() ? "" : frBelPin.getBELName() + "/") + frBelPin.getName();
+                    BELPin toBelPin = outputPin.getSiteConns().get(0);
+                    String toBelPinString = (toBelPin.isSitePort() ? "" : toBelPin.getBELName() + "/") + toBelPin.getName();
+                    Short delayPs = intrasiteAndLogicDelayModel.getIntraSiteDelay(siteInst.getSiteTypeEnum(), frBelPinString, toBelPinString);
+                    if (delayPs != null && delayPs > 0) {
+                        DeviceResources.Device.CornerModel.Builder delayBuilder = spBuilder.initDelay();
+                        DeviceResources.Device.CornerModelValues.Builder slowBuilder = delayBuilder.initSlow().initSlow();
+                        slowBuilder.initMax().setMax(delayPs * 1e-12f);
```

Comment:
Perhaps pull out `1e-12f` and other occurrences into a separate static variable with a note.